src/modules/home/components/reservation-card.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useState, useEffect } from 'react';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import TextField from '@mui/material/TextField';
+import Button from '@mui/material/Button';
+import MenuItem from '@mui/material/MenuItem';
+import Divider from '@mui/material/Divider';
+
+interface ReservationCardProps {
+  pricePerNight: number;
+  cleaningFee: number;
+  serviceFee: number;
+  maxGuests: number;
+  availableFrom: string;
+  availableTo: string;
+}
+
+export const ReservationCard: React.FC<ReservationCardProps> = ({
+  pricePerNight,
+  cleaningFee,
+  serviceFee,
+  maxGuests,
+  availableFrom,
+  availableTo,
+}) => {
+  const [checkIn, setCheckIn] = useState<string>('');
+  const [checkOut, setCheckOut] = useState<string>('');
+  const [guests, setGuests] = useState<number>(1);
+  const [isButtonEnabled, setIsButtonEnabled] = useState<boolean>(false);
+
+  const getTodayDate = (): string => {
+    const today = new Date();
+    const year = today.getFullYear();
+    const month = String(today.getMonth() + 1).padStart(2, '0');
+    const day = String(today.getDate()).padStart(2, '0');
+    return `${year}-${month}-${day}`;
+  };
+
+  const calculateTotal = (): number => {
+    if (checkIn && checkOut) {
+      const start = new Date(checkIn);
+      const end = new Date(checkOut);
+      const nights = Math.max(0, (end.getTime() - start.getTime()) / (1000 * 60 * 60 * 24));
+      return nights * pricePerNight + cleaningFee + serviceFee;
+    }
+    return 0;
+  };
+
+  const calculateNights = (): number => {
+    if (checkIn && checkOut) {
+      const start = new Date(checkIn);
+      const end = new Date(checkOut);
+      return Math.max(0, (end.getTime() - start.getTime()) / (1000 * 60 * 60 * 24));
+    }
+    return 0;
+  };
+
+  const validateDates = (): boolean => {
+    if (checkIn && checkOut) {
+      const start = new Date(checkIn);
+      const end = new Date(checkOut);
+      const availableStart = new Date(availableFrom);
+      const availableEnd = new Date(availableTo);
+
+      return start >= availableStart && end <= availableEnd && start < end;
+    }
+    return false;
+  };
+
+  useEffect(() => {
+    setIsButtonEnabled(validateDates());
+  }, [checkIn, checkOut]);
+
+  const todayDate = getTodayDate();
+
+  return (
+    <Box
+      sx={{
+        border: '1px solid #ddd',
```

Comment:
Please extract styles that have more than 2 properties into ```styles``` constant here and in other parts